packages/admin/cms-admin/src/common/MasterMenuRoutes.tsx

```diff
@@ -33,18 +34,24 @@ export function useRoutePropsFromMasterMenuData(items: MasterMenuData): RoutePro
 
 export interface MasterMenuRoutesProps {
     menu: MasterMenuData;
+    fallback?: ReactNode;
 }
 
-export const MasterMenuRoutes = ({ menu }: MasterMenuRoutesProps) => {
+export const MasterMenuRoutes = ({ menu, fallback = <NotFound /> }: MasterMenuRoutesProps) => {
     const routes = useRoutePropsFromMasterMenuData(menu);
     const match = useRouteMatch();
-
     return (
         <Switch>
             <Redirect to={`${match.url}${routes[0].path}`} exact={true} from={match.path} />
             {routes.map((route, index) => (
                 <RouteWithErrorBoundary key={index} {...route} path={`${match.path}${route.path}`} />
             ))}
+            <RouteWithErrorBoundary
+                component={() => {
+                    return fallback;
+                }}
+                path="*"
+            />
```

Comment:
Optional: We could switch to the [recommended children rendering](https://v5.reactrouter.com/web/api/Route/route-render-methods).
```tsx
<RouteWithErrorBoundary
  path="*"
>
  {fallback}
</RouteWithErrorBoundary>
```